src/internal/connector/graph/service.go

```diff
@@ -173,6 +173,8 @@ type clientConfig struct {
 	// The minimum delay in seconds between retries
 	minDelay           time.Duration
 	overrideRetryCount bool
+
+	appendMiddleware []khttp.Middleware
```

Comment:
very minor nit: `additionalMiddleware` feels like a better fit.

Reply:
I like `append` because it can be very important to know whether the middleware gets called at the beginning of the stack or the end of it.